### PR TITLE
config: change the default value of `enable-dynamic-config` to true and make it as a command argument

### PR DESCRIPTION
**Pull request by qw4990**
### What problem does this PR solve? 
Change the default value of `enable-dynamic-config` to true and make it as a command argument.

### Check List 

Tests 

 - Unit test

### CODE REVIEW

File: tidb-server/main.go [hunk 1]
```diff
@@ -69,45 +69,47 @@ import (
...
-	nmTokenLimit       = "token-limit"
-	nmPluginDir        = "plugin-dir"
-	nmPluginLoad       = "plugin-load"
-	nmRepairMode       = "repair-mode"
-	nmRepairList       = "repair-list"
+	nmVersion             = "V"
+	nmConfig              = "config"
+	nmConfigCheck         = "config-check"
+	nmConfigStrict        = "config-strict"
+	nmEnableDynamicConfig = "enable-dynamic-config"
```
Comment (qw4990) [inline tidb-server/main.go]: A new variable is added here.

File: tidb-server/main.go [hunk 2]
```diff
@@ -69,45 +69,47 @@ import (
...
 var (
-	version      = flagBoolean(nmVersion, false, "print version information and exit")
-	configPath   = flag.String(nmConfig, "", "config file path")
-	configCheck  = flagBoolean(nmConfigCheck, false, "check config file validity and exit")
-	configStrict = flagBoolean(nmConfigStrict, false, "enforce config file validity")
+	version             = flagBoolean(nmVersion, false, "print version information and exit")
+	configPath          = flag.String(nmConfig, "", "config file path")
+	configCheck         = flagBoolean(nmConfigCheck, false, "check config file validity and exit")
+	configStrict        = flagBoolean(nmConfigStrict, false, "enforce config file validity")
+	enableDynamicConfig = flagBoolean(nmEnableDynamicConfig, true, "enable update configs dynamically")
```
Comment (overvenus) [inline tidb-server/main.go]:
```suggestion
	enableDynamicConfig = flagBoolean(nmEnableDynamicConfig, true, "Enable dynamic configuration change")
```

File: config/config.go [hunk 3]
```diff
@@ -631,7 +631,7 @@ var defaultConf = Config{
 	Experimental: Experimental{
 		AllowAutoRandom: false,
 	},
-	EnableDynamicConfig: false,
```
Comment (rleungx) [inline config/config.go]: The default value should be changed to `false`

Comment (qw4990) [reply config/config.go]: fixed. PTAL @rleungx

**Comment by siddontang** [comment]
don't add a new flag, we must remove this flag in PD and TiKV too.

**Comment by qw4990** [comment]
> don't add a new flag, we must remove this flag in PD and TiKV too.

OK, I will close this PR soon.

**Comment by qw4990** [comment]
Close due to that we don't need this feature.

**Pull request closed without merging**